sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -343,6 +397,44 @@ public void finish(@Nullable SpanStatus status) {
   @Override
   @ApiStatus.Internal
   public void finish(@Nullable SpanStatus status, @Nullable SentryDate finishDate) {
+
+    // auto-finish any open spans
+    for (Span span : getSpans()) {
```

Comment:
`h`: as discussed internally, maybe let's try to create a new type for this kind of spans instead of introducing `SpanOptions` and packing everything into `SentryTracer`. There may be 2 different ways to solve this going forward:
* try to have `IdleSpan` with `idleTimeout` which finishes itself after some timeout. This would require changing Activity txs to `trimEnd = true` to have the correct end timestamp for them
* try to have `IdleSpan` without the idle timeout, so we could just iterate here the same way we do now, and check if the span is instance of `IdleSpan`, and force-finish it. It would then do the trimStart/trimEnd logic inside its own `finish` method.

Reply:
@romtsn Sounds good, but I'm wondering how far I should go here? Spans right now are a half-baked POJOs, e.g. a span can not directly access it's children. On the other hand a span has access to it's root transaction.

Reply:
hmm, good point. Would you mind asking about that in `team-client-infra` to see if anyone has concerns if we keep track of child spans in Span and not only in SentryTracer? For me it would be fine, but maybe there's something we should align on, or at least document.

Also, there's a [SpanRecorder](https://github.com/getsentry/sentry-javascript/blob/f8047f66ac8abf64f992b5d681007383ea175276/packages/tracing/src/span.ts#L18-L40) class available on JS. Not sure if it's exactly what we want to achieve, but maybe we could reuse some bits.